drivers/sensor/nxp/nxp_tmpsns/nxp_tmpsns.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2025 Polytech A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tmpsns
+
+#include <fsl_tempsensor.h>
+#include <zephyr/drivers/sensor.h>
+
+struct nxp_tmpsns_data {
+	float die_temp;
+};
+
+static int nxp_tmpsns_channel_get(const struct device *dev, enum sensor_channel chan,
+				  struct sensor_value *val)
+{
+	const struct nxp_tmpsns_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ALL:
```

Comment:
Delete this line.
`get` cannot support `SENSOR_CHAN_ALL` since there is only one `val` reference.